src/Command/InfectionCommand.php

```diff
@@ -368,58 +371,65 @@ private function checkMetrics(): bool
 
     private function initContainer(InputInterface $input): void
     {
-        /** @var string|null $configFile */
-        $configFile = $input->hasOption('configuration')
-            ? trim((string) $input->getOption('configuration'))
-            : null
-        ;
+        // Currently the configuration is mandatory hence there is no way to
+        // say "do not use a config". If this becomes possible in the future
+        // though, it will likely be a `--no-config` option rather than relying
+        // on this value to be set to an empty string.
+        $configFile = trim((string) $input->getOption('configuration'));
 
-        if ($configFile === '') {
-            $configFile = null;
+        $coverage = trim((string) $input->getOption('coverage'));
+        $testFramework = trim((string) $this->input->getOption('test-framework'));
+        $testFrameworkOptions = trim((string) $this->input->getOption('test-framework-options'));
+        $initialTestsPhpOptions = trim((string) $input->getOption('initial-tests-php-options'));
+
+        $minMsi = $input->getOption('min-msi');
+
+        if (null !== $minMsi && $minMsi !== (string) (float) $minMsi) {
+            throw new InvalidArgumentException(sprintf('Expected min-msi to be a float. Got "%s"', $minMsi));
         }
 
-        /** @var string|null $mutators */
-        $mutators = $input->hasOption('mutators')
-            ? trim((string) $input->getOption('mutators'))
-            : null
-        ;
+        $minCoveredMsi = $input->getOption('min-covered-msi');
 
-        if ($mutators === '') {
-            $mutators = null;
+        if (null !== $minCoveredMsi && $minCoveredMsi !== (string) (float) $minCoveredMsi) {
+            throw new InvalidArgumentException(sprintf('Expected min-covered-msi to be a float. Got "%s"', $minCoveredMsi));
         }
 
+        $mutators = trim((string) $input->getOption('mutators'));
+
         $this->container = $this->getApplication()->getContainer()->withDynamicParameters(
```

Comment:
There is actually really not much changes here. The main differences are:

- The `Configuration` still returns `null` for a couple of properties: this gives greater flexibility for the future to take advantage of the `null` value to say infer a configuration (e.g. pick a default or guess a default based on other values). The only downside was that I had to add a couple of casting in some places for BC, but that's it so I think it's ok
- I simplified some of the expressions (no semantic change) e.g.:

```php
$x = $input->hasOption('foo') ? trim((string) $input->getOption('foo')) : null;

if ('' === $x) {
  $x = null;
}

// use $x
```

into:

```php
$x = trim((string) $input->getOption('foo'));

// when using $x
'' === $x ? null : $x;
```

And:

```
(bool) $input->getOption('foo')
```

I removed the casting where unnecessary